Sources/NostrSDK/Events/ParameterizedReplaceableEvent.swift

```diff
@@ -22,4 +22,20 @@ public extension ParameterizedReplaceableEvent {
 
         return try? EventCoordinates(kind: kind, pubkey: publicKey, identifier: identifier, relayURL: relayURL)
     }
+
+    func shareableEventCoordinates(relayURLStrings: [String]? = nil, excludeAuthor: Bool = false, excludeKind: Bool = false) throws -> String {
+        let validatedRelayURLStrings = try relayURLStrings?.map {
```

Comment:
Since this block of code is nearly identifier between parameterized and non-parameterized replaceable events, you could factor out this code into a helper function inside an extension of the `ReplaceableEvent` protocol. It would need to accept one more argument for `identifier`.

That way, here (and without the identifier in the non-parameterized variant) you could simply call:
```swift
try shareableEventCoordinates(relayURLStrings: relayURLStrings,
                                      excludeAuthor: excludeAuthor,
                                      excludeKind: excludeKind,
                                      identifier: identifier ?? "")
```